tsdb/wlog/watcher_test.go

```diff
@@ -171,22 +177,31 @@ func TestTailSamples(t *testing.T) {
 
 				for j := 0; j < histogramsCount; j++ {
 					inner := rand.Intn(ref + 1)
+					hist := &histogram.Histogram{
+						Schema:          2,
+						ZeroThreshold:   1e-128,
+						ZeroCount:       0,
+						Count:           2,
+						Sum:             0,
+						PositiveSpans:   []histogram.Span{{Offset: 0, Length: 1}},
+						PositiveBuckets: []int64{int64(i) + 1},
+						NegativeSpans:   []histogram.Span{{Offset: 0, Length: 1}},
+						NegativeBuckets: []int64{int64(-i) - 1},
+					}
+
 					histogram := enc.HistogramSamples([]record.RefHistogramSample{{
 						Ref: chunks.HeadSeriesRef(inner),
 						T:   now.UnixNano() + 1,
-						H: &histogram.Histogram{
-							Schema:          2,
-							ZeroThreshold:   1e-128,
-							ZeroCount:       0,
-							Count:           2,
-							Sum:             0,
-							PositiveSpans:   []histogram.Span{{Offset: 0, Length: 1}},
-							PositiveBuckets: []int64{int64(i) + 1},
-							NegativeSpans:   []histogram.Span{{Offset: 0, Length: 1}},
-							NegativeBuckets: []int64{int64(-i) - 1},
-						},
+						H:   hist,
 					}}, nil)
+
 					require.NoError(t, w.Log(histogram))
+					floatHistogram := enc.FloatHistogramSamples([]record.RefFloatHistogramSample{{
```

Comment:
We are not checking if we watched the float histograms in this test. There is `expectedHistograms` for integer histograms; we need one for float histograms.